sql/core/src/test/scala/org/apache/spark/sql/execution/WholeStageCodegenSuite.scala

```diff
@@ -325,4 +326,68 @@ class WholeStageCodegenSuite extends QueryTest with SharedSQLContext {
       checkAnswer(groupByWithId, Seq(Row(1, 2, 0), Row(1, 2, 0)))
     }
   }
+
+  test("WholeStageCodegen does not work properly for LocalTableScanExec") {
+    // Case1: LocalTableScanExec is the root of a query plan tree.
+    // In this case, WholeStageCodegenExec should not be inserted
+    // as the direct parent of LocalTableScanExec.
+    val df1 = spark.createDataset(1 to 10).toDF
+    val rootOfExecutedPlan = df1.queryExecution.executedPlan
+
+    // Ensure WholeStageCodegenExec is not inserted and
+    // LocalTableScanExec is still the root.
+    assert(!rootOfExecutedPlan.isInstanceOf[WholeStageCodegenExec],
+      "WholeStageCodegenExec should not be inserted if LocalTableScanExec is the only plan.")
+    assert(rootOfExecutedPlan.isInstanceOf[LocalTableScanExec],
+      "LocalTableScanExec should be still the root.")
+
+    // Case2: The parent of a LocalTableScanExec supports WholeStageCodegen.
+    // In this case, the LocalTableScanExec should be within a WholeStageCodegen domain
+    // and no more InputAdapter is inserted as the direct parent of the LocalTableScanExec.
+    val leftDF = spark.createDataset(1 to 10).toDF
+    val rightDF = spark.createDataset(1 to 10).toDF
+
+    // Force BroadcastHasJoin enabled
+    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> Long.MaxValue.toString) {
```

Comment:
nit: How about using `HashAggregate` for the test? If so, we could remove this config set;
```
scala> Seq(1, 2, 3).toDF("a").groupBy().sum().explain
== Physical Plan ==
*(2) HashAggregate(keys=[], functions=[sum(cast(a#52 as bigint))])
+- Exchange SinglePartition
   +- *(1) HashAggregate(keys=[], functions=[partial_sum(cast(a#52 as bigint))])
      +- LocalTableScan [a#52]
```

Reply:
Replaced with HashAggregate.